From ff476d7eb7eb4acf6dd6681a3464bf308359ae3e Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Fri, 5 Dec 2025 11:47:23 +0100
Subject: [PATCH 07/17] mmc: atmel-sdhci: add support for LAN969x

LAN969x is compatible with the existing driver, however it requires the
GCK clock rate to be set to 100000000 Hz instead.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 drivers/mmc/Kconfig       | 2 +-
 drivers/mmc/atmel_sdhci.c | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/mmc/Kconfig
+++ b/drivers/mmc/Kconfig
@@ -565,7 +565,7 @@ config MMC_SDHCI_ASPEED
 
 config MMC_SDHCI_ATMEL
 	bool "Atmel SDHCI controller support"
-	depends on ARCH_AT91
+	depends on ARCH_AT91 || ARCH_MICROCHIPSW
 	depends on MMC_SDHCI
 	help
 	  This enables support for the Atmel SDHCI controller, which supports
--- a/drivers/mmc/atmel_sdhci.c
+++ b/drivers/mmc/atmel_sdhci.c
@@ -15,6 +15,7 @@
 
 #define ATMEL_SDHC_MIN_FREQ	400000
 #define ATMEL_SDHC_GCK_RATE	240000000
+#define LAN969X_GCK_RATE	100000000
 
 #define ATMEL_SDHC_MC1R 0x204
 #define ATMEL_SDHC_MC1R_FCD	0x80
@@ -169,6 +170,7 @@ static const struct udevice_id atmel_sdh
 	{ .compatible = "atmel,sama5d2-sdhci", .data = ATMEL_SDHC_GCK_RATE },
 	{ .compatible = "microchip,sam9x60-sdhci", .data = ATMEL_SDHC_GCK_RATE },
 	{ .compatible = "microchip,sama7g5-sdhci", .data = ATMEL_SDHC_GCK_RATE },
+	{ .compatible = "microchip,lan9691-sdhci", .data = LAN969X_GCK_RATE },
 	{ }
 };
 
